engine/src/main/scala/cromwell/services/KeyValueServiceActor.scala

```diff
@@ -0,0 +1,72 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptor
+import cromwell.engine.db.DataAccess
+import cromwell.services.KeyValueServiceActor._
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object KeyValueServiceActor {
+  trait KvMessage
+  trait KvAction extends KvMessage with ServiceRegistryMessage {
+    def serviceName = "KeyValue"
+  }
+  trait KvResponse extends KvMessage
+
+  case class KvPut(jobDescriptor: BackendJobDescriptor, key: String, value: Option[String]) extends KvAction
```

Comment:
Appreciate the thought. :)

Still, current Cromwell data access convention is to use the singleton, so until a full replacement arrives, I'm fine with continuing to use the singleton.